packages/core/src/components/steps-modal/steps-modal.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  forceUpdate,
+  h,
+} from '@stencil/core'
+
+@Component({
+  tag: 'atom-steps-modal',
+  styleUrl: 'steps-modal.scss',
+  shadow: false,
+})
+export class AtomStepsModal {
+  @Prop() steps: number
+  @Prop({ mutable: true }) currentStep = 1
+  @Prop() trigger?: string
+  @Prop() stepsTitles: string
+  @Prop({ mutable: true }) isOpen = false
+
+  @Event() atomFinish: EventEmitter
+  @Event() atomCancel: EventEmitter
+  @Event() atomNextStep: EventEmitter
+  @Event() atomPreviousStep: EventEmitter
+  @Event() atomCloseClick: EventEmitter
+  @Event() atomDidDismiss: EventEmitter
+  @Event() atomDidPresent: EventEmitter
+
+  @Element() el!: HTMLElement
+
+  private stepsTitlesArray: string[] = []
+
+  componentWillLoad() {
+    if (
+      this.currentStep > this.steps ||
+      this.currentStep < 1 ||
+      this.currentStep === undefined
+    ) {
+      this.currentStep = 1
+    }
+
+    this.stepsTitlesArray = this.stepsTitles.split(',')
+  }
+
+  private handleStep = (step: number) => {
+    this.currentStep = step
+  }
+
+  private handlePrimaryClick = () => {
+    if (this.currentStep === this.steps) {
+      this.atomFinish.emit()
+    } else {
+      this.handleStep(this.currentStep + 1)
+      this.atomNextStep.emit(this.currentStep)
+      forceUpdate(this.el)
+    }
+  }
```

Comment:
suggestion: we could improve a bit it

```suggestion
  private handlePrimaryClick = () => {
    const { currentStep, steps, atomFinish, atomNextStep, el } = this;

    currentStep === steps
      ? atomFinish.emit()
      : (this.handleStep(currentStep + 1), atomNextStep.emit(currentStep), forceUpdate(el));
  }
```